extensions/firefox/content/PdfStreamConverter.jsm

```diff
@@ -1013,6 +1013,7 @@ PdfStreamConverter.prototype = {
 
     // Keep the URL the same so the browser sees it as the same.
     channel.originalURI = aRequest.URI;
+    channel.loadInfo.resultPrincipalURI = aRequest.loadInfo.resultPrincipalURI;
```

Comment:
> Existence of the new property on nsILoadInfo depends on landing [1].

Given that there's also a Firefox addon using this code-base, please see https://github.com/mozilla/pdf.js#firefox-and-seamonkey, the code needs to remain backwards compatible.

Is it possible to easily feature detect this, using e.g. something like the following code?
```javascript
if ("resultPrincipalURI" in aRequest.loadInfo) {
  channel.loadInfo.resultPrincipalURI =
    aRequest.loadInfo.resultPrincipalURI;
}
```